### PR TITLE
caddyhttp: Allow matching Transfer-Encoding, add to access logs

### PR DESCRIPTION
**Pull request by francislavoie**
Fixes https://github.com/caddyserver/caddy/issues/6628

### CODE REVIEW

File: modules/caddyhttp/matchers.go [hunk 1]
```diff
@@ -967,22 +967,26 @@ func (MatchHeader) CELLibrary(_ caddy.Context) (cel.Library, error) {
...
-func getHeaderFieldVals(input http.Header, fieldName, host string) []string {
+// The host parameter should be obtained from the http.Request.Host field, and the
+// transferEncoding from http.Request.TransferEncoding, since net/http removes them
+// from the header map.
+func getHeaderFieldVals(input http.Header, fieldName, host string, transferEncoding []string) []string {
 	fieldName = textproto.CanonicalMIMEHeaderKey(fieldName)
 	if fieldName == "Host" && host != "" {
 		return []string{host}
 	}
+	if fieldName == "Transfer-Encoding" && input[fieldName] == nil {
```
Comment (mohammed90) [inline modules/caddyhttp/matchers.go:978]: Is the check `input[fieldName] == nil` necessary if we know Go stdlib always moves the `Transfer-Encoding` header out of the header map?

Comment (francislavoie) [reply modules/caddyhttp/matchers.go:978]: If some `header` directive set it for some reason, it could be set I guess.

Comment (mohammed90) [reply modules/caddyhttp/matchers.go:978]: Yeah, that's possible, I guess.

**Review by mohammed90 — APPROVED** [review]
LGTM

**Review by mholt — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**